File: workspace/people/p03-security-grc-analyst/home/downloads/inbox/2026-07-20/automation/2026-07-20-intake-evidence-package-validation-tool.py
# spine_traceability: org.cobalt-harbor=Cobalt Harbor Systems | audit.ar-260713-027=AR-260713-027 fictional audit request for AC-27 review material | date.2026-07-20=2026-07-20 closure-evidence follow-up date
"""Synthetic AC-27 intake validator; no network or filesystem operations."""

RECORD = {"artifact_id": "p03-full-005166", "audit_request": "AR-260713-027", "case": "CASE-260713-184", "control": "AC-27", "disposition": "synthetic intake normalized; bounded review retained", "evidence_record": "EV-184-19", "evidence_set": 184, "network_note": "No live endpoints, raw events, or routable addresses are represented.", "organization": "Cobalt Harbor Systems", "program": "Aegis Assurance", "record_id": "INB-090", "review_date": "2026-07-20", "risk": "RISK-031", "source_id": "p03-src-005166", "unique_marker": "foundation-b001-090"}

def validate(record: dict[str, object]) -> bool:
    required = {"control", "audit_request", "case", "evidence_set", "risk"}
    return required.issubset(record) and record["control"] == "AC-27" and record["evidence_set"] == 184

if __name__ == "__main__":
    assert validate(RECORD), "synthetic intake record is incomplete: foundation-b001-090"

File: workspace/people/p03-security-grc-analyst/home/downloads/inbox/2026-07-20/automation/2026-07-20-intake-evidence-package-validation-tool.rs
pub fn review_summary() -> Vec<(&'static str, &'static str)> {
    vec![
    ("artifact_id", "p03-full-006990"),
    ("organization", "Cobalt Harbor"),
    ("risk_reference", "RISK-031"),
    ("review_date", "2026-07-20"),
    ("evidence_question", "What decision can be made from the retained source material without external access?"),
    ("evidence_instrument", "asset accountability list"),
    ("review_observation", "the local conclusion did not depend on a portal or live endpoint"),
    ("challenge_performed", "a scope boundary was compared with the evidence collection boundary"),
    ("review_profile", "the evidence set was collected before the approval meeting; the evidence custodian confirmed the record boundary; the reviewer sampled an adverse operating condition; the source fact supported acceptance with a documented caveat; the comparison used the prior review conclusion; the evidence was retained in the review register; the exception path was considered and recorded; the reviewer found no handover impact; the decision was suitable for targeted follow-up; the source chronology contained a reconciled late entry; the control claim was independently corroborated; the review boundary identified unavailable material as a limitation; the next assurance trigger is completion of the follow-up action; the disposition was communicated to the owner; the retained rationale supports repeat review; the record remains a local offline assurance artifact"),
    ("counterexample", "A dated narrative without a named custodian is insufficient because responsibility cannot be tested."),
    ("decision_rule", "Record an approval only after the reviewer can reproduce the conclusion from the retained facts without network access."),
    ("disposition", "accepted with a retained review note"),
    ("operating_boundary", "No Kio capability is used or required for this offline review.")
    ]
}
